### DISCUSSION TITLE
Define the basic idea of ​​the system.

### DISCUSSION THREAD

**Discussion by louies0623** [helloSystem/hello#406] [General]
I want to try 5W1H to target.

1. Who is using
2. Why use this system
3. What is the use of this system
4. When will it be used
5. Where is this system used
6. How to do

**Comment by louies0623** [comment]
I give an example windows, you can also think together.
1. Those who touch the computer for the first time know that there is a moving arrow that can control the computer.
2. because it is easy to use.
3. Copy the software given by the school teacher on the desktop, because he has no internet to use.
4. He's doing a simple poster and it's due tomorrow.
5. He lives in a rural area of ​​Korea and uses an old laptop.
6. He took the flash drive and inserted it into the laptop, he moved the software to the desktop (so that the next installation will be more stable), clicked on the installation software, and after the completion of the installation, he simply entered a few Korean characters and used the mouse to draw,  After saving, copy his completed file from the desktop to the pen drive, and complete.

**Comment by probonopd** [comment]
Hello @louies0623, the basic idea is that you can sit someone who has been using Macs from 1984 to ~2010 (before it started to get iOS'ized) in front of a helloSystem machine and they feel right at home. Without anything being really identical, but conceptually similar enough to make people "feel at home". They should be able to just use the machine without any special instructions.

**Reply by louies0623** [reply]
Exactly, This is what today's open source systems lack.

**Reply by louies0623** [reply]
Makes it feel like the computer is saying "Hello! I'm your easier work assistant" .

**Comment by louies0623** [comment]
From the above example, when comparing helloSystem, The current problem with helloSystem is

1. Support some laptops.
2. Language and input method support.
3. A method or package that enables easy installation of software.
4. Interfaces that can be controlled by a mouse (eg: login screen, settings).
5. How to install fonts
6. Whether the font size of the system is appropriate or too small.（Because the screen of a small laptop is very small）
7. The background program with high CPU usage, but the performance of the real program is too low.

**Reply by probonopd** [reply]
Hello @louies0623 . Please open indvidual tickets for each of them with _details_. Without that, nothing can be improved because I don't know what the actual issues are.

Support some laptops. -- > I am running this on the [Framework Laptop](https://frame.work/de/de/products/laptop-diy-11-gen-intel)
Language and input method support. --> It uses German on my system because I have a German [Raspberry Pi Keyboard and Hub](https://www.raspberrypi.com/products/raspberry-pi-keyboard-and-hub/) attached. No configuration needed, autodetected
A method or package that enables easy installation of software. --> The system has "placeholder applications" that ask the user to download the software when needed (e.g., LibreOffice)
Interfaces that can be controlled by a mouse (eg: login screen, settings). -->  System -> Applications -> Preferences
How to install fonts --> System -> Applications -> Preferences -> Fonts
Whether the font size of the system is appropriate or too small.（Because the screen of a small laptop is very small）--> Please provide details: Your screen's resolution in pixels, and in inches
The background program with high CPU usage, but the performance of the real program is too low. --> Please provide details. Which background program has high CPU usage?

**Reply by louies0623** [reply]
First, I have four laptops, but none of them can execute your helloSystem.
Second, your system does not correctly display the language I use and the input method that can switch languages ​​on my three desktop computers, and unnecessary key swapping.
Third, can you be sure that everyone will only use the software built into the system to work, such as Darktable, Poedit, VLC, etc. If you add every open source software to it, you will have a long list of menus, even  There are categories, but you have to search one by one slowly to find what you want. (For example, Blender is not in the 3D category, this weird category) is too troublesome; this is the importance of the package, recommend Appimage or Flatpak, otherwise add it  app store.

**Reply by louies0623** [reply]
Fourth, what should a user do if he wants to play an offline or non-steam game.
Fifth, the login screen can only be controlled by the keyboard and the basic settings also need to be modified by the terminal, such as the region, date and time settings. You said that you only need to connect to the Internet or set the efi in the computer. Yes, you are right, but only  It is correct on your side, because the internal time design of the system is the GMT time on your side rather than the GMT time in the user's computer, which needs to be set, but where to set it.
Sixth, on the 10.1" screen of the eee pc shell series (my relative's), the English characters appear very small, let alone foreign languages, and it will be squeezed into a ball.

**Reply by probonopd** [reply]
> First, I have four laptops, but none of them can execute your helloSystem.

Please give details about each system. Only then I can possibly help.

> Second, your system does not correctly display the language I use and the input method that can switch languages ​​on my three desktop computers, and unnecessary key swapping.

Setting the language is not implemented yet in the way I want to implement it. We are not at 1.0 yet ;-)

> Third, can you be sure that everyone will only use the software built into the system to work, such as Darktable, Poedit, VLC, etc. If you add every open source software to it, you will have a long list of menus

There will be a high bar for each category. Ideally there will be no more than 1-2 carefully selected applications in each category. So that users not familiar with open source applications know immediately which one to use.

> Fourth, what should a user do if he wants to play an offline or non-steam game.

Play an offline or non-steam game.

> Fifth, the login screen can only be controlled by the keyboard and the basic settings also need to be modified by the terminal, such as the region, date and time settings. 

Can you please elaborate on the login screen? I don't even see it, as the main user is logged in automatically by default. For region, date and time settings my objective is to make these fully automatic so that we never have to fiddle with those manualy, ever. For me it already works :) but more can be done in this area.

> Sixth, on the 10.1" screen of the eee pc shell series (my relative's), the English characters appear very small

Please open a ticket with details, including the exact model of the computer, the screen size in inches and in pixels.

Thank you very much for your ideas!

**Reply by grahamperrin** [reply]
> … four laptops, but none of them can execute your helloSystem. …

If you mean that none of them can boot the operating system: this might be an issue with FreeBSD that was quite widespread, but relatively understated. 

The fix was outlined at [2021 in Review: Software Development | FreeBSD Foundation](https://freebsdfoundation.org/blog/2021-in-review-software-development/) under _Improved amd64 UEFI boot_. In a little more detail, amongst May 2022 release notes for FreeBSD-13.1-RELEASE: 

* <https://www.freebsd.org/releases/13.1R/relnotes/#boot-loader>

In context: <https://github.com/helloSystem/ISO/releases/latest> _latest_ helloSystem 0.7.0 is based on FreeBSD 13.0-RELEASE, which lacks the fix, and reached its end of life around three months ago. 

<https://www.freebsd.org/security/unsupported/>

**Reply by probonopd** [reply]
Currently we are working toward  helloSystem 0.8.0 which is based on FreeBSD 13.1-RELEASE. Experimental builds are available.

**Comment by grahamperrin** [comment]
> I want to try 5W1H …

What is _5W1H_?

**Comment by louies0623** [comment]
>  What is 5W1H?

There is an explanation.[5W1H ](https://datamyte.com/5w1h-method-comprehensive-guide/) 
Otherwise there is another approach. 
- [shneiderman 8 golden rules](https://www.cs.umd.edu/users/ben/goldenrules.html), [Guide2](https://www.interaction-design.org/literature/article/shneiderman-s-eight-golden-rules-will-help-you-design-better-interfaces)
- [Nielsen and Molich's 10 User Interface Design Guidelines](https://www.interaction-design.org/literature/article/user-interface-design-guidelines-10-rules-of-thumb)

**Reply by grahamperrin** [reply]
> There is an explanation.[5W1H ](https://datamyte.com/5w1h-method-comprehensive-guide/) 

Thanks, I requested a correction to the page. 

(The _No Comments_ link seems to have no effect; no interface to add a comment.)

**Reply by louies0623** [reply]
https://datamyte.com/5w1h-method-comprehensive-guide/

https://www.webomates.com/blog/software-testing/software-testing-and-the-5w-and-1h-technique-of-metrics-reporting/

**Comment by louies0623** [comment]
This is the second example: system errors, automation time zones, and inconveniences that users can't handle on their own
•	What is the problem?
The time zone cannot be set
•	Where is the problem?
There is no option to adjust the time zone, datetime, and time
•	When is the problem?
On some pages on the browser, "NET::ERR_CERT_DATE_INVALID" will appear and the time needs to be adjusted
•	Why is the problem?
Because the automatic time zone caused by the system time is not the user's local time zone (he will understand if he takes a plane to use it in a foreign country)
•	Who is affected by the problem?
Users traveling abroad and users who do not live in German Standard Time
•	How can we solve the problem?
Add settings

**Reply by probonopd** [reply]
Did you find any German timezone hardcoded? That would be a bug.
The clean solution would probably be to introduce separate EFI variables for keyboard layout, language, locale (do we need both?), country, timezone. Did I miss any?
What is important for me is that these are per-machine, not per-OS settings. Because I don't want to fiddle around with these on each Live ISO.

**Reply by louies0623** [reply]
Usually the browser can perceive the built-in time zone of the current system. I use this to prove that the time of the virtual computer in Virtualbox is consistent with the time of the entity, but helloOS is not, and why it is not the CMOS of the personal computer when entering the system  The time is fine, and the subsequent correction is on the setting.

**Reply by louies0623** [reply]
![think](https://user-images.githubusercontent.com/44593430/204124924-9a514017-4196-48ae-9b09-d9ce7c01d287.png)

**Reply by probonopd** [reply]
Possibly we should skip setting the timezone if we are running inside a virtual machine?

In any case, if you find any hardcoded German anywhere, then it's a bug.

**Reply by louies0623** [reply]
![think](https://user-images.githubusercontent.com/44593430/204125267-1ce476f4-59a5-48e5-bd66-99448f1bf152.png)

**Reply by probonopd** [reply]
Please answer:
* Is your computer hardware clock set to UTC internally? (yes/no)
* Is your virtual machine hardware clock set to UTC? (yes/no)

This is one of the ongoing hassles in computing, really frustrating. Can't we all standardize on either using or not using UTC for the hardware clocks...

**Reply by grahamperrin** [reply]
> time zone cannot be set

Related: 

* https://github.com/helloSystem/Utilities/issues/37 (2021-01-10)
* https://github.com/helloSystem/ISO/issues/224 (2021-06-23)
* https://github.com/helloSystem/Utilities/issues/75 (2021-07-01)

**Reply by louies0623** [reply]
> Is your computer hardware clock set to UTC internally?

I don't know. 

> Is your virtual machine hardware clock set to UTC?

 yes

And on the physical machine is also the wrong time zone as shown above.
UPDATE: Turning off UTC in the virtual machine does the same.

**Reply by probonopd** [reply]
> I don't know.

Me neither... and that is part of the mess that setting times on computers with multiple OSes and/or VM brings.

**Reply by louies0623** [reply]
I was thinking, first use the CMOS BIOS time as the system default display time...

**Reply by probonopd** [reply]
That works only if it is set to local time, not to UTC.

**Reply by louies0623** [reply]
Adding settings may be able to know what's going on with this system.

**Reply by probonopd** [reply]
According to https://discussions.apple.com/thread/1986443

> Windows and Unix-based operating systems handle the hardware clock differently: Windows assumes it runs on local time, while Linux and MacOS assume it runs on UTC. Obviously, in a multi-boot situation this will create mayhem and confusion.

helloSystem being a Unix-based operating system, we should assume it runs on UTC.

**Reply by louies0623** [reply]
It would be great if you could avoid Y2k38 or similar Unix UTC time bugs.

**Reply by probonopd** [reply]
Well. That's probably something to be handled in FreeBSD, which helloSystem is built upon.

**Reply by louies0623** [reply]
At present, in all operating systems, you can give some examples that do not need to go through the time zone setting during the installation process, and its time zone time is correct. Maybe you can refer to.

I didn’t find it, because Windows has it, Mac also has it to create users when it starts for the first time, and the same is true for android, not to mention fedora, ubuntu, how about you.

**Reply by probonopd** [reply]
Yes, I am sick and tired of having to set the timezone each time I start a Live ISO. Hence, the timezone gets automatically set in helloSystem. For example, if a German keyboard is attached, then the matching timezone for Germany gets set. Unfortunately this does not work in all cases, as there is not always a 1:1 relationship between a language, keyboard layout, and timezone. Therefore, the best solution will probably be to introduce a new EFI NVRAM variable for the timezone. So that one can set it one time and then never have to think about it again. Ideally, all operating systems would use that same EFI NVRAM variable.

**Reply by louies0623** [reply]
Try it, or use the CMOS time of the computer during Live, and set the time zone after the first startup after installation.

**Reply by louies0623** [reply]
I found a problem this time, that is, one country has one time zone, and some countries are very large and span several time zones, so how do you deal with it?
![image](https://user-images.githubusercontent.com/44593430/208920993-aa35497f-32fc-43af-ac35-32c99c30736e.png)
![image](https://user-images.githubusercontent.com/44593430/208920587-67e9a781-6b36-4f76-ab87-47d8d5324529.png)
![image](https://user-images.githubusercontent.com/44593430/208920847-47f6aede-14e4-4caa-b1b9-3c064399929e.png)
![image](https://user-images.githubusercontent.com/44593430/208924053-d41cd857-a3e9-477f-b32d-3488e33ba56c.png)

**Reply by louies0623** [reply]
I recently discovered that the location of the hello OS installer time zone on the live cd is correct, but the system is wrong, and the system is still in the wrong time zone after installation.
![thinka_工作區域 1](https://user-images.githubusercontent.com/44593430/209457112-0d542e04-2138-4ba6-9f18-a65800c53bbb.png)

**Comment by louies0623** [comment]
This is the third example: Desktop computers are vulnerable to data theft or tampering without a screen lock function
• What's the question?
Screen lock required to protect data
• where is the problem?
no lock screen option
• When did the problem arise?
There is no way to lock it when you temporarily leave the computer to go to the bathroom
• Why is there a problem?
There is no lock screen in the system
• Who is affected by the problem?
Librarians, counter personnel, design companies, product developers, product design users, report personnel, teachers, business managers, financial professionals, secretaries, parents with children, personnel in the personnel department, government agencies, etc.
• How do we solve problems?
Add lock screen

**Reply by probonopd** [reply]
Lock screens were not a part of Classic Mac until well after System 7. They are not essential for a personal computer. They can be easily circumvented too, unless all data is encrypted. Which we might implement at some point in the future.

**Reply by louies0623** [reply]
hm... in the end, Is still need this function.

**Reply by louies0623** [reply]
@probonopd If you go to the hospital and find out that they use hello os, then the system is not locked, and your body and medical records are easily taken away

**Reply by probonopd** [reply]
helloSystem is for _personal_ computers. I would not recommend a hospital to use it in semi-public places.

**Reply by louies0623** [reply]
I have a friend whose relatives came to visit. The relative’s son used my friend’s mobile phone when his mobile phone was not locked, and used my friend’s account to leave messages on the Internet. As a result, he was blocked by the community, and it was you and you  How does your computer feel.

**Reply by probonopd** [reply]
Auto-login into slim can be disabled, but currently it needs to be done by editing `/usr/local/etc/slim.conf`.

**Reply by louies0623** [reply]
And what I want is to temporarily leave to protect the current work progress, not to log out of that kind of lock！
